---
title: "PCA analysis of the Mosquito dataset using adegenet"
output: word_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
load some libraries

```{r}
library("adegenet")
library ("dplyr")

```



```{r}
library(dplyr)
file.rename("batch_1.genepop",  "batch_1_genepop.gen")
```
import the dataset and convert the SNP file into a genind object

```{r}
mosquito_snp <- read.genepop("batch_1_genepop.gen")
mosquito_snp
```

PCA analysis



6.2 Performing a Principal Component Analysis on genind objects
The tables contained in genind objects can be submitted to a Principal Component Analysis
(PCA) to seek a summary of the genetic diversity among the sampled individuals. Such
analysis is straightforward using adegenet to prepare data and ade4 for the analysis per
se. One has first to extract allele counts or frequencies from the genind object and replace
missing data (NAs) by the mean allele frequency. This is achieved by tab:
```{r}

paste("There are", sum(is.na(mosquito_snp$tab)), "missing values that need to be replaced. They will all be replaced with scaleGen:")


```

```{r}
X <- scaleGen(mosquito_snp, NA.method="mean")
class(X)
dim(X)
```
Note that alternatively, we could have used na.replace to replace missing data, and then
left the centring/scaling to dudi.pca.
The analysis can now be performed. We disable the scaling in dudi.pca, which would
erase the scaling choice made earlier in scaleGen. Note: in practice, retained axes can be
chosen interactively by removing the arguments scannf=FALSE,nf=3.

```{r}
pca1 <- dudi.pca(X,cent=FALSE,scale=FALSE,scannf=FALSE,nf=3)
barplot(pca1$eig,main="PCA eigenvalues", col=heat.colors(50))
```
```{r}
pca1
```
The basic scatterplot for this analysis can be obtained by:
```{r}
s.label(pca1$li)
title("PCA of mosquito dataset\naxes 1-2")
add.scatter.eig(pca1$eig, 3,1,2)
```
However, this figure can largely be improved. First, we can use s.class to represent both
the genotypes and inertia ellipses for populations.
```{r}
s.class(pca1$li, pop(mosquito_snp))
title("PCA of mosquito dataset\naxes 1-2")
add.scatter.eig(pca1$eig, 3,1,2)
```
This plane shows that the main structuring is between African an French breeds, the second
structure reflecting genetic diversity among African breeds. The third axis reflects the
diversity among French breeds:

```{r}
s.class(pca1$li,pop(mosquito_snp),xax=1,yax=3,sub="PCA 1-3",csub=2)
title("PCA of mosquito dataset\naxes 1-3")
add.scatter.eig(pca1$eig,nf=3,xax=1,yax=3)
```
However, we can yet improve these scatterplots, which are fortunately easy to customize.
For instance, we can remove the grid, choose different colors for the groups, use larger dots
and transparency to better assess the density of points, and remove internal segments of the
ellipses:

```{r}
col <- funky(15)
s.class(pca1$li, pop(mosquito_snp),xax=1,yax=3, col=transp(col,.6), axesell=FALSE,
cstar=0, cpoint=3, grid=FALSE)
```
Let us now assume that we ignore the group memberships. We can still use color in an
informative way. For instance, we can recode the principal components represented in the
scatterplot on the RGB scale:
```{r}
colorplot(pca1$li, pca1$li, transp=TRUE, cex=3, xlab="PC 1", ylab="PC 2")
title("PCA of mosquito dataset\naxes 1-2")
abline(v=0,h=0,col="grey", lty=2)
```

Colors are based on the first three PCs of the PCA, recoded respectively on the red, green,
and blue channel. In this figure, the genetic diversity is represented in two complementary
ways: by the distances (further away = more genetically different), and by the colors (more
different colors = more genetically different).
We can represent the diversity on the third axis similarly:

```{r}
colorplot(pca1$li[c(1,3)], pca1$li, transp=TRUE, cex=3, xlab="PC 1", ylab="PC 3")
title("PCA of mosquito dataset\naxes 1-3")
abline(v=0,h=0,col="grey", lty=2)
```


